polkadot/zombienet-sdk-tests/tests/smoke/coretime_revenue.rs

```diff
@@ -0,0 +1,508 @@
+
+	let _subscriber = tokio::spawn(async move {
+		para_watcher(p_api, p_events).await;
+	});
+
+	let api: OnlineClient<PolkadotConfig> = para_node.wait_client().await?;
+	let _s1 = tokio::spawn(async move {
+		ti_watcher(api, "PARA").await;
+	});
+	let api: OnlineClient<PolkadotConfig> = relay_node.wait_client().await?;
+	let _s2 = tokio::spawn(async move {
+		ti_watcher(api, "RELAY").await;
+	});
+
+	log::info!("Initiating teleport from RC's account of Alice to PC's one");
+
+	// Teleport some Alice's tokens to the Coretime chain. Although her account is pre-funded on
+	// the PC, that is still neccessary to bootstrap RC's `CheckedAccount`.
+	relay_client
+		.tx()
+		.sign_and_submit_default(
+			&rococo_api::tx().xcm_pallet().teleport_assets(
+				VersionedLocation::V4(Location {
+					parents: 0,
+					interior: Junctions::X1([Junction::Parachain(1005)]),
+				}),
+				VersionedLocation::V4(Location {
+					parents: 0,
+					interior: Junctions::X1([Junction::AccountId32 {
+						network: None,
+						id: alice.public_key().0,
+					}]),
+				}),
+				VersionedAssets::V4(Assets(vec![Asset {
+					id: AssetId(Location { parents: 0, interior: Junctions::Here }),
+					fun: Fungibility::Fungible(1_500_000_000),
+				}])),
+				0,
+			),
+			&alice,
+		)
+		.await?;
+
+	wait_for_para_event(
+		para_events.clone(),
+		"Balances",
+		"Minted",
+		|e: &coretime_api::balances::events::Minted| e.who == alice_acc,
+	)
+	.await;
+
+	// RC's total issuance doen't change, but PC's one increses after the teleport
+
+	total_issuance.1 += 1_500_000_000;
+	assert_total_issuance(relay_client.clone(), para_client.clone(), total_issuance).await;
+
+	log::info!("Initializing broker and starting sales");
+
+	// Initialize broker and start sales
+
+	para_client
+		.tx()
+		.sign_and_submit_default(
+			&coretime_api::tx().sudo().sudo(CoretimeRuntimeCall::Utility(
+				CoretimeUtilityCall::batch {
+					calls: vec![
+						CoretimeRuntimeCall::Broker(CoretimeBrokerCall::configure {
+							config: BrokerConfigRecord {
+								advance_notice: 5,
+								interlude_length: 1,
+								leadin_length: 1,
+								region_length: 1,
+								ideal_bulk_proportion: Perbill(100),
+								limit_cores_offered: None,
+								renewal_bump: Perbill(10),
+								contribution_timeout: 5,
+							},
+						}),
+						CoretimeRuntimeCall::Broker(CoretimeBrokerCall::request_core_count {
+							core_count: 3,
+						}),
+						CoretimeRuntimeCall::Broker(CoretimeBrokerCall::set_lease {
+							task: 1005,
+							until: 1000,
+						}),
+						CoretimeRuntimeCall::Broker(CoretimeBrokerCall::start_sales {
+							end_price: 45_000_000,
+							extra_cores: 2,
+						}),
+					],
+				},
+			)),
+			&alice,
+		)
+		.await?;
+
+	log::info!("Waiting for a full-length sale to begin");
+
+	// Skip the first sale completeley as it may be a short one. Also, `request_code_count` requires
+	// two session boundaries to propagate. Given that the `fast-runtime` session is 10 blocks and
```

Comment:
You should be able to wait for an event emitted at a session change on the relay. If you wait for two and then go to the sale after it, that means that the relative timings are less important.